File: protocol/registration.rs
use serde::{Deserialize, Serialize};

use crate::protocol::{global_definitions::ModuleType, module_event::ModuleEvent};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub struct Registration {
    pub id: String,
    pub module_type: ModuleType,
    pub lool_up_id: String,
    pub parent_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(tag = "type", content = "payload")]
pub enum ProtocolMessage {
    Registration(Registration),
    ModuleEvent(ModuleEvent),
}

File: UI/src/lib/protocol/registration.ts
import { z } from "zod";
import { ModuleTypeSchema } from "./module-type";

export const RegistrationSchema = z.object({
    id: z.string(),
    module_type: ModuleTypeSchema,
    lool_up_id: z.string(),
    parent_id: z.string(),
});

export type Registration = z.infer<
    typeof RegistrationSchema
>;